### PR TITLE
Responsive InfoDialog for AMO

### LINKED ISSUE #1414: Show InfoDialog when an add-on is installed
The mobile frontend will be able to install add-ons once #1388 lands. The data is being set in state for the `InfoDialog` to appear but since it isn't being rendered and doesn't support mobile.

Update `InfoDialog` to support mobile, move it to `core/` and render it in `App` so that it can be displayed on install.

### PR DESCRIPTION
**Pull request by mstriemer**
info-dialog-responsive mov

Fixes #1414.

### CODE REVIEW

File: src/core/reducers/addons.js [hunk 1]
```diff
@@ -26,6 +26,7 @@ export default function addon(state = initialState, action) {
       } else if (thisAddon.current_version && thisAddon.current_version.files.length > 0) {
         newState[key] = {
           ...thisAddon,
+          iconUrl: thisAddon.icon_url,
```
Comment (mstriemer) [inline src/core/reducers/addons.js]: This is quite unforunate but I looked into switching everything over to camel cased and it looks like a good amount of work. I filed #1536.

File: src/core/containers/InfoDialog/index.js [hunk 2]
```diff
@@ -39,4 +41,19 @@ export class InfoDialogBase extends React.Component {
   }
 }
 
-export default translate()(onClickOutside(InfoDialogBase));
+export const InfoDialogBase = compose(
```
Comment (tofumatt) [inline src/core/containers/InfoDialog/index.js]: 👍

File: src/disco/containers/DiscoPane.js [hunk 3]
```diff
@@ -87,7 +85,7 @@ export class DiscoPaneBase extends React.Component {
   }
 
   render() {
-    const { results, i18n, AddonComponent, showInfoDialog, infoDialogData } = this.props;
+    const { results, i18n, AddonComponent } = this.props;
```
Comment (tofumatt) [inline src/disco/containers/DiscoPane.js]: While this is being changed, `AddonComponent` should be first.

**Comment by mstriemer** [comment]
> Why the move/rename of src/core/components/InfoDialog/index.js → src/core/containers/InfoDialog/index.js?

I moved it because it now handles its own redux state. I thought that was our distinction between components and containers. I think the distinction was inspired by https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0 originally but after reading it again it probably makes more sense to move the base stuff into `ui/components/InfoDialog` and make this `core/containers/InstallDialog`. I'm not sure that would provide much benefit right now though. I still think it makes sense for it to be a container but maybe we're not on the same page about that distinction.

**Comment by mstriemer** [comment]
I guess I could have put the `ShowInfoDialog` bit into `core/containers/InfoDialog` and left the base stuff alone in the components folder.

**Pull request merged**